extractor.py

```diff
@@ -1,146 +1,186 @@
 #!/usr/bin/env python3
-
 from argparse import ArgumentParser
+from json import JSONDecodeError
+import config
+import json
 import mimetypes
 import os
 
-DEBUG = False
-
-class Program:
-    _tar_type = 'application/x-tar'
-    _gzip_type = 'gzip'
-    _zip_type = 'application/zip'
-    _7z_type = 'application/x-7z-compressed'
-
-    _tar_ext = '.tar'
-    _targz_ext = '.tar.gz'
-    _gzip_ext = '.gz'
-    _zip_ext = '.zip'
-    _7z_ext = '.7z'
-
-    _tar_prog = 'tar'
-    _targz_prog = 'tar'
-    _gzip_prog = 'guzip'
-    _zip_prog = 'unzip'
-    _7z_prog = '7z'
-    
-    _tar_attr = 'xvf'
-    _targz_attr = 'xzvf'
-    _gzip_attr = ''
-    _zip_attr = ''
-    _7z_attr = 'x'
-
-    exts = {
-            _tar_type: _tar_ext,
-            _gzip_type: _gzip_ext,
-            _zip_type: _zip_ext,
-            _7z_type: _7z_ext,
-    }
-
-    progs = {
-            _tar_ext: _tar_prog,
-            _targz_ext: _targz_prog,
-            _gzip_ext: _gzip_prog,
-            _zip_ext: _zip_prog,
-            _7z_ext: _7z_prog,
-    }
+DEBUG = True
+
+_attrs_str = 'attrs'
+_configs_str = 'configs'
+_encoding_str = 'encoding'
+_ext_str = 'ext'
+_result_str = 'result'
+_message_str = 'message'
+_error_str = 'error'
+_error_code_str = 'error_code'
+_prog_str = 'prog'
+
+def _logd(*args, sep=' ', end='\n', file=None):
+    if DEBUG:
+        print(args, sep, end, file)
+
+def _error_builder(error_code, error_message):
+    return {_error_code_str: error_code, _message_str: error_message}
+
+def _result_builder(result_status, prog_for_ext):
+    return {_result_str: result_status, _prog_str: prog_for_ext}
+
+def _response_builder(error_obj, result_obj, message):
+    return {_error_str: error_obj, _result_str: result_obj, _message_str: message}
+
+def _check_extension(ext, encoding):
+    if not ext and not encoding:
+        _logd('check_extension, ext and encoding are null')
+        err = _error_builder(1, 'Null extension and encoding')
+        return _response_builder(err, None, err[_message_str])
+    _logd('check_extension:', ext, encoding)
+
+    if not os.path.isfile(config.EXTRACTOR_CONFIG_FILE):
+        _logd('No file', config.EXTRACTOR_CONFIG_FILE)
+        return _response_builder(None, _result_builder(False, None), 'No file configured')
+
+    with open(config.EXTRACTOR_CONFIG_FILE, 'r') as f:
+        try:
+            json_data = json.load(f)
+        except JSONDecodeError:
+            _logd('Could not get load configurations')
+            err = _error_builder(2, 'File is not json')
+            return _response_builder(err, None, err[_message_str])
+
+    _logd('file content:', json_data)
+
+    try:
+        configs = json_data[_configs_str]
+    except:
+        msg = 'File with different json object than expected' 
```

Comment:
here you didn't follow your _ convention for the "msg" variable.. there are some other instances of this happening

Reply:
I learned that the underscore convention is used to represent private elements of a module or class, so I removed it all in the variables created inside of a function. The only instance of this happening is the for in line 62 where I broke this convention to make the underscore represent the variables inside of that for scope (I don't know if it makes sense though).

The other references to _something are all for private function callings or module private variables.